Installation issue: MED

### Steps to reproduce the issue

```console
$ spack spec -I 'med@4.1.1 fflags=-fdefault-integer-8 ~doc+fortran+python+shared'
 -   med@4.1.1%gcc@13.3.0 fflags=-fdefault-integer-8 +api23~doc+fortran~ipo+mpi+python+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu24.04-skylake
[+]      ^cmake@3.30.5%gcc@13.3.0~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release patches=dbc3892 arch=linux-ubuntu24.04-skylake
[+]          ^curl@8.10.1%gcc@13.3.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu24.04-skylake
[+]              ^nghttp2@1.63.0%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^ncurses@6.5%gcc@13.3.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu24.04-skylake
[+]          ^zlib-ng@2.2.1%gcc@13.3.0+compat~new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu24.04-skylake
[+]      ^gcc-runtime@13.3.0%gcc@13.3.0 build_system=generic arch=linux-ubuntu24.04-skylake
[e]      ^glibc@2.39%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[e]      ^gmake@4.3%gcc@13.3.0~guile build_system=generic patches=599f134 arch=linux-ubuntu24.04-skylake
[+]      ^hdf5@1.10.7%gcc@13.3.0~cxx~fortran~hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make patches=2a1e311,f42732a arch=linux-ubuntu24.04-skylake
[e]          ^pkgconf@1.8.1%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[e]      ^openmpi@4.1.6%gcc@13.3.0+atomics~cuda+cxx~cxx_exceptions~debug~gpfs~internal-hwloc~internal-libevent~internal-pmix+java~legacylaunchers~lustre~memchecker~openshmem~orterunprefix+pmi+romio+rsh~singularity~static~two_level_namespace+vt~wrapper-rpath build_system=autotools fabrics=ofi,psm,psm2,ucx romio-filesystem=none schedulers=slurm arch=linux-ubuntu24.04-skylake
[+]      ^python@3.13.0%gcc@13.3.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=linux-ubuntu24.04-skylake
[+]          ^bzip2@1.0.8%gcc@13.3.0~debug~pic+shared build_system=generic arch=linux-ubuntu24.04-skylake
[e]              ^diffutils@3.10%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^expat@2.6.4%gcc@13.3.0+libbsd build_system=autotools arch=linux-ubuntu24.04-skylake
[+]              ^libbsd@0.12.2%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]                  ^libmd@1.0.4%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^gdbm@1.24%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[e]          ^gettext@0.21%gcc@13.3.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^libffi@3.4.6%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^openssl@3.4.0%gcc@13.3.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu24.04-skylake
[+]              ^ca-certificates-mozilla@2023-05-30%gcc@13.3.0 build_system=generic arch=linux-ubuntu24.04-skylake
[e]              ^perl@5.38.2%gcc@13.3.0~cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu24.04-skylake
[+]          ^readline@8.2%gcc@13.3.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu24.04-skylake
[+]          ^sqlite@3.46.0%gcc@13.3.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^util-linux-uuid@2.40.2%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^xz@5.4.6%gcc@13.3.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu24.04-skylake
[+]      ^swig@4.1.1%gcc@13.3.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]          ^pcre2@10.44%gcc@13.3.0~jit+multibyte build_system=autotools arch=linux-ubuntu24.04-skylake
```


### Error message

<details><summary>Error message</summary>

<pre>
             _MEDfieldInfo(PyObject*, PyObject*, PyObject*)':
  >> 3861    /tmp/bmarchand/spack-stage/spack-stage-med-4.1.1-xefkbvhckoi2utsazeqa64hndllhozbf/spack-build-xefkbvh/python/med/medfield_modulePYTHON_wrap.cxx:24360:14: error: 'PyEval_Call
             Object' was not declared in this scope; did you mean 'PyObject_CallObject'?
     3862    24360 |     pinst  = PyEval_CallObject(pclass, pargs);
     3863          |              ^~~~~~~~~~~~~~~~~
     3864          |              PyObject_CallObject
     3865    /tmp/bmarchand/spack-stage/spack-stage-med-4.1.1-xefkbvhckoi2utsazeqa64hndllhozbf/spack-build-xefkbvh/python/med/medfield_modulePYTHON_wrap.cxx: In function 'PyObject* _wrap
             _MEDfieldInfoByName(PyObject*, PyObject*, PyObject*)':
  >> 3866    /tmp/bmarchand/spack-stage/spack-stage-med-4.1.1-xefkbvhckoi2utsazeqa64hndllhozbf/spack-build-xefkbvh/python/med/medfield_modulePYTHON_wrap.cxx:25023:14: error: 'PyEval_Call
             Object' was not declared in this scope; did you mean 'PyObject_CallObject'?
</pre></details>


### Information on your system

* **Spack:** 0.23.1 (2bfcc69fa870d3c6919be87593f22647981b648a)
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-skylake

### Additional information

Hello @likask ; the problem comes from `PyEval_CallObject` deprecated since Python 3.9 and removed in Python 3.13. Is it possible to add a constraint on Python version in the MED recipe ? 


### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/med/package.py
+++ b/repos/spack_repo/builtin/packages/med/package.py
@@ -118,7 +118,7 @@ class Med(CMakePackage):
         if "+mpi" in spec:
             options.extend(["-DMEDFILE_USE_MPI=YES", "-DMPI_ROOT_DIR=%s" % spec["mpi"].prefix])
 
-        if "+python" in spec and spec['python'].version >= Version('3.9'):
+        if "+python" in spec and spec["python"].version >= Version("3.9"):
             options.extend(["-DCMAKE_CXX_FLAGS=-DPyEval_CallObject=PyObject_CallObject"])
 
         return options

--- a/repos/spack_repo/builtin/packages/med/package.py
+++ b/repos/spack_repo/builtin/packages/med/package.py
@@ -118,4 +118,7 @@ class Med(CMakePackage):
         if "+mpi" in spec:
             options.extend(["-DMEDFILE_USE_MPI=YES", "-DMPI_ROOT_DIR=%s" % spec["mpi"].prefix])
 
+        if "+python" in spec and spec['python'].version >= Version('3.9'):
+            options.extend(["-DCMAKE_CXX_FLAGS=-DPyEval_CallObject=PyObject_CallObject"])
+
         return options

